### PR TITLE
Modify chainDeal params

### PR DESCRIPTION
**Pull request by Ugo**

### CODE REVIEW

File: src/main/java/com/iexec/common/chain/DealParams.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+@Getter
+public class DealParams {
+
+    // Note to dev: the naming of the variables here is important since it will be stored on-chain
+    // that is why the names are not java compliant (use of underscore and no upper case)
+    private String iexec_args;
+    private List<String> iexec_input_files;
```
Comment (zguesmi) [inline src/main/java/com/iexec/common/chain/DealParams.java]: You can use iexecArgs and iexecInputFiles, the mapping is done automatically by ObjectMapper.

File: src/main/java/com/iexec/common/chain/ChainDeal.java [hunk 2]
```diff
@@ -48,16 +48,20 @@
...
         try {
-            LinkedHashMap tasksParamsMap = new ObjectMapper().readValue(params, LinkedHashMap.class);
-            listParams = new ArrayList<String>(tasksParamsMap.values());
+            DealParams dealParams = new ObjectMapper().readValue(params, DealParams.class);
+            if(dealParams.getIexec_input_files() == null) {
+                dealParams.setIexec_input_files(new ArrayList<>());
+            }
+            return dealParams;
         } catch (IOException e) {
             log.warn("Params string is not a JSON, considering the string is one full param");
```
Comment (james-toussaint) [inline src/main/java/com/iexec/common/chain/ChainDeal.java]: You can remove this log (to many occurrences in prod)

**Review by james-toussaint — APPROVED** [review]

File: src/main/java/com/iexec/common/chain/DealParams.java (hunk 1 above)
Comment (Ugo) [reply src/main/java/com/iexec/common/chain/DealParams.java]: The mapping is not working if iexecArgs and iexecInputFiles are used.

Comment (zguesmi) [reply src/main/java/com/iexec/common/chain/DealParams.java]: I think the @JsonProperty is needed.

Comment (Ugo) [reply src/main/java/com/iexec/common/chain/DealParams.java]: ok good with the proper JsonProperty

File: src/main/java/com/iexec/common/chain/ChainDeal.java (hunk 2 above)
Comment (Ugo) [reply src/main/java/com/iexec/common/chain/ChainDeal.java]: done

**Pull request merged**